src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/Object.WriteTests.cs

```diff
@@ -146,5 +149,75 @@ public static void WriteObjectWithNumberHandling()
             var options = new JsonSerializerOptions { NumberHandling = JsonNumberHandling.AllowReadingFromString };
             JsonSerializer.Serialize(new object(), options);
         }
+
+        /// <summary>
+        /// Also see <see cref="Utf8JsonWriterTests.WriteLargeJsonToStreamWithoutFlushing"/>
+        /// </summary>
+        [PlatformSpecific(TestPlatforms.Windows | TestPlatforms.OSX)]
+        [ConditionalFact(nameof(Utf8JsonWriterTests.IsX64))]
+        [OuterLoop]
+        public static void SerializeLargeListOfObjects()
+        {
+            Dto dto = new()
+            {
+                Prop1 = int.MaxValue,
+                Prop2 = int.MinValue,
+                Prop3 = "AC",
+                Prop4 = 500,
+                Prop5 = int.MaxValue / 2,
+                Prop6 = 250.5M / 3,
+                Prop7 = 250.5M / 3,
+                Prop8 = 250.5M / 3,
+                Prop9 = 250.5M / 3,
+                Prop10 = 250.5M / 3,
+                Prop11 = 150.6M / 3,
+                Prop12 = 150.6M / 3,
+                Prop13 = DateTimeOffset.Now,
+                Prop14 = DateTimeOffset.Now,
+                Prop15 = DateTimeOffset.Now,
+                Prop16 = DateTimeOffset.Now,
+                Prop17 = 3,
+                Prop18 = DateTime.Now,
+                Prop19 = DateTime.Now,
+                Prop20 = 25000,
+                Prop21 = DateTime.Now
+            };
+
+            List<Dto> items = Enumerable.Repeat(dto, 3_000_000).ToList();
+
+            try
+            {
+                JsonSerializer.Serialize(items);
```

Comment:
Maybe use `Assert.Throws` instead or add a failing assert after this line to make sure that this does indeed throw.